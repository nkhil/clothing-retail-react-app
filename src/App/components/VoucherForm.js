import React, { PureComponent, Fragment } from "react"

class VoucherForm extends PureComponent {
  discountVoucherRef = React.createRef()

  handleSubmit = e => {
    e.preventDefault()
  }

  render = () => {
    return (
      <Fragment>
        <form id="voucher-form" onSubmit={this.handleSubmit}>
          <input
            type="text"
            name="discount-code"
            ref={this.discountVoucherRef}
            placeholder="Voucher code"
          />
          <button type="submit">Submit</button>
        </form>
      </Fragment>
    )
  }
}

export default VoucherForm
