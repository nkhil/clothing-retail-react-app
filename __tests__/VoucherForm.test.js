import "@babel/polyfill"
import React from "react"
import VoucherForm from "../src/App/components/VoucherForm"
import renderer from "react-test-renderer"
import ReactDOM from "react-dom"

require("react-router-dom")
jest.mock("react-router-dom")

test("VoucherForm renders without crashing", () => {
  const div = document.createElement("div")
  ReactDOM.render(<VoucherForm />, div)
})

test("VoucherForm renders correctly", () => {
  const voucherForm = renderer.create(<VoucherForm />).toJSON()
  expect(voucherForm).toMatchSnapshot()
})
