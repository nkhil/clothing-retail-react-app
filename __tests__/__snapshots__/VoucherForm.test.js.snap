// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VoucherForm renders correctly 1`] = `
<form
  id="voucher-form"
  onSubmit={[Function]}
>
  <input
    name="discount-code"
    placeholder="Voucher code"
    type="text"
  />
  <button
    type="submit"
  >
    Submit
  </button>
</form>
`;
